lib/HttpSignatureStrategy.js

```diff
@@ -134,24 +133,39 @@ Strategy.prototype.authenticate = function(req) {
       }
       callback();
     }],
-    verify: ['validatePublicKey', function(callback, results) {
-      var verified;
-      try {
-        verified = httpSignature.verifySignature(
-          results.parseRequest, results.getPublicKey.publicKeyPem);
-      } catch(ex) {
-        return callback(new BedrockError(
-          'Request signature verify error.',
-          'HttpSignature.VerifyError', {cause: ex}));
+    verify: ['validatePublicKey', (results, callback) => {
+      const {publicKeyBase58, publicKeyPem} = results.getPublicKey;
+      const {signature} = results.parseRequest.params;
+      const {algorithm, signingString: plaintext} = results.parseRequest;
+      const verifyOptions = {plaintext, signature};
+      if(jsonld.hasValue(
+        results.getPublicKey, 'type', 'Ed25519VerificationKey2018')) {
+        verifyOptions.algorithm = 'ed25519';
+        verifyOptions.publicKeyBase58 = publicKeyBase58;
       }
-      if(!verified) {
-        return callback(new BedrockError(
-          'Request signature verification failed.',
-          'HttpSignature.VerifyFailure'));
+      if(jsonld.hasValue(
+        results.getPublicKey, 'type', 'RsaVerificationKey2018')
+        && algorithm.startsWith('RSA')) {
```

Comment:
Well, we can't exactly ignore the specified algorithm because the hashType must be extracted from it.  But we can certainly remove `algorithm` from this conditional.

Also this does bring up the topic of backwards compatibility.  Are we going with 'the only way is forward' here?

Reply:
The hashType for an `RsaVerificationKey2018` is SHA-256. It isn't permitted to be anything else.

Reply:
Btw, don't release a new version of bedrock-passport anytime soon, I've got several other other changes that need to go into the next version that are unrelated to http signatures (at least not directly related).